guide/creating-your-bot/command-handling.md

```diff
@@ -115,60 +115,91 @@ client.commands = new Collection();
-const { clientId, guildId, token } = require('./config.json');
+With the correct files identified, the last step is to loop over the array and dynamically set each command into the `client.commands` Collection. In addition to the filter to remove non-JavaScript files, here it also checks that the files being loaded all have at least the `data` and `execute` properties. This helps to prevent errors resulting from loading empty, unfinished or otherwise incorrect command files while you're still developing.
 
-const commands = [];
-const commandsPath = path.join(__dirname, 'commands');
-const commandFiles = fs.readdirSync(commandsPath).filter(file => file.endsWith('.js'));
+## Receiving command interactions
 
-for (const file of commandFiles) {
-	const filePath = path.join(commandsPath, file);
-	const command = require(filePath);
-	commands.push(command.data.toJSON());
-}
+Every slash command is an `interaction`, so to respond to a command, you need to create a listener for the `interactionCreate` event that will execute code when your application receives an interaction. Place the code below in the `index.js` file we created earlier.
+
+```js
+client.on('interactionCreate', interaction => {
+	console.log(interaction);
+});
 ```
 
-## Dynamically executing commands
+However, not every interaction is a slash command (e.g. `MessageComponent`s). Make sure to only receive and handle slash commands in this function by making use of the `BaseInteraction#isChatInputCommand()` method to exit if another type is encountered:
 
-You can use the `client.commands` Collection setup to retrieve and execute your commands! Inside the `interactionCreate` event, delete the `if`/`else if` chain of commands and replace it with this:
+```js {2}
+client.on('interactionCreate', interaction => {
+	if (!interaction.isChatInputCommand()) return;
+	console.log(interaction);
+});
+```
+
+## Executing commands
+
+::: tip
+This section assumes you're using the `client.commands` convention from the [loading commands](/creating-your-bot/loading-commands.md) page of this guide. Please carefully read those first so that you can understand the methods used in this section.
+:::
 
-```js {4-13}
+When your bot receives an `interactionCreate` event, the interaction object contains all the information you need to dynamically retrieve and execute your commands!
+
+Let's take a look at our `ping` command. Note the `execute` method that will reply to the interaction with "Pong!".
+
+```js
+module.exports = {
+	data: new SlashCommandBuilder()
+		.setName('ping')
+		.setDescription('Replies with Pong!'),
+	async execute(interaction) {
+		await interaction.reply('Pong!');
+	},
+};
+```
+
+First, you need to get the matching command from the `client.commands` Collection based on the `interaction.commandName`. Your Client instance is always available via `interaction.client`. If no matching command is found, log an error to the console and ignore the event.
```

Comment:
`Client` (code, if referring to the class, maybe even as docs link) or client (lower)